apps/aecore/src/aec_db.erl

```diff
@@ -1052,6 +1096,85 @@ handle_activity_result(Res, ErrorKeys) ->
             exit(Err)
     end.
 
+do_activity(transaction, Fun) ->
+    try
+    mnesia:activity(transaction, fun() ->
+        R = Fun(),
+        case persistent_term:get(?BYPASS, no_bypass) of
+            {rocksdb, Handles} ->
+                {mnesia, _, {tidstore, TStore, _, _}} = get(mnesia_activity_state),
```

Comment:
The TidStore table is a bag, and can contain sequences of writes/deletes to the same key. Thus, order is critical. At least at first glance, the walk seems to reverse the order of updates.

Reply:
Sorry, mnesia already resolves this before commit.